src/learn/database/index.md

```diff
@@ -0,0 +1,688 @@
+  SELECT
+    id,
+    content,
+    TIME(created_at)
+  FROM tasks
+`);
+```
+
+If you log the result of this query you should see the task objects change:
+
+```json
+[{ "id": 2, "content": "Send mum flowers", "TIME(created_at)": "08:52:30" }]
+```
+
+However the column name has changed to represent this. Ideally we would keep the same name (`created_at`). We can rename columns using the `AS` SQL operator:
+
+```js
+const select_tasks = db.prepare(/*sql*/ `
+  SELECT
+    id,
+    content,
+    TIME(created_at) AS created_at
+  FROM tasks
+`);
+```
+
+If you run this again you should see the object key is now `created_at` like before.
+
+```json
+[{ "id": 2, "content": "Send mum flowers", "created_at": "08:52:30" }]
+```
+
+## Deleting a row
+
+Let's add another model function that can delete a task from our `tasks` table. It should take an ID parameter and delete just the row that matches.
+
+```js
+const delete_task = db.prepare(/*sql*/ `
+  DELETE FROM tasks WHERE id = ?
+`);
+
+function removeTask(id) {
+  return delete_task.run(id);
+}
+```
+
+{% box "error" %}
+
+SQL queries act on **every row** by default. This makes deletion and updates very dangerous. Without a `WHERE` clause or other condition they will delete/update _every_ row.
+
+{% endbox %}
+
+Test this by calling `removeTasks(1)` to delete the first task you created earlier. You can call `listTasks` to check that a task has been removed.
+
+## Testing the model
+
+So far we've been manually testing things by calling our model functions. It would be better to write some automated tests to make this reproducible and reusable. That way we can catch reversions if the code breaks later on.
+
+Let's write a test to verify several of our model functions. Create a new file `test/tasks.test.js`:
+
+```js
+const test = require("node:test");
+const assert = require("node:assert");
+const model = require("../model/tasks.js");
+const db = require("../database/db.js");
+
+test("can create, remove & list tasks", () => {
+  db.exec("DELETE FROM tasks");
+
+  const task = model.createTask("test task");
+  assert.equal(task.id, 1);
+  assert.equal(task.content, "test task");
+  assert.equal(task.complete, 0);
+
+  model.removeTask(task.id);
+  const tasks = model.listTasks();
+  assert.equal(tasks.length, 0);
+});
+```
+
+We can run the test directly with:
+
+```shell
+node test/tasks.test.js
+```
+
+Or we can run all our tests with:
+
+```shell
+node --test
+```
+
+Note we are **not** setting the `DB_FILE` environment variable. This means our DB will be _in-memory_ for our tests, so none of the changes will be persisted or effect the dev DB we've been using.
+
+{% box %}
+
+**Important**: we empty the `tasks` table at the start of each test to ensure any leftovers from other tests don't effect this one. This ensures tests are not dependent on each other, so they can be run in any order (or in parallel) without affecting the results.
+
+{% endbox %}
+
+## Updating a row
```

Comment:
SUGGESTION: Maybe move to before Section on DELETE TASKS, as (I think) no task left otherwise? This would allow it to be done before moving on to Testing.

If so, move "Toggling a Boolean section" with it?

Maybe explicitly provide example of the code to run this - although maybe this is handholding too much?

Reply:
I've emphasised that they should run the `seed.js` script after deleting the DB before, so they should have those 3 tasks (2 after deleting one of them).

I've also added an explicit unit test to this section (and delete) so they can check it that way rather than manually.